docs/tagging.md

```diff
@@ -0,0 +1,64 @@
+| CSIVolumeName                   | pvcName                   | CSIVolumeName = pvc-a3ab0567-3a48-4608-8cb6-4e3b1485c808                 | Added to all volumes.                                                                                                                                                                                                                             |
+| CSIVolumeSnapshotName           | volumeSnapshotContentName | CSIVolumeSnapshotName = snapcontent-69477690-803b-4d3e-a61a-03c7b2592a76 | Added to all snapshots.                                                                                                                                                                                                                           |
+| fsx.openzfs.csi.aws.com/cluster | true                      | fsx.openzfs.csi.aws.com/cluster = true                                   | Add to all volumes, snapshots, and final backups. This allows users to track which resources were created via the FSx for OpenZFS CSI driver. Users may also use a policy to limit the csi driver's permissions to just the resources it manages. |                                                                                      |
+
+# StorageClass Tagging
+
+The AWS FSx for OpenZFS CSI Driver supports tagging through `StorageClass.parameters`
+
+**Example 1**
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: fsx-openzfs-sc
+provisioner: fsx.openzfs.csi.aws.com
+parameters:
+  ...
+  ...
+  tags: "key1=value1,key2=hello world,key3="
+```
+
+Provisioning a volume using this StorageClass will apply the following tags:
+
+```
+key1=value1
+key2=hello world
+key3=<empty string>
+```
+
+# Snapshot Tagging
+The AWS FSx for OpenZFS CSI Driver supports tagging snapshots through `VolumeSnapshotClass.parameters`, similar to StorageClass tagging.
+
+**Example**
+```
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshotClass
+metadata:
+  name: csi-aws-vsc
+driver: fsx.openzfs.csi.aws.com
+deletionPolicy: Delete
+parameters:
+  tags: "key1=value1,key2=hello world,key3="
+```
+
+Provisioning a snapshot in namespace 'ns-prod' with `VolumeSnapshot` name being 'fsx-openzfs-snapshot' using this VolumeSnapshotClass, will apply the following tags to the snapshot:
+
+```
+key1=value1
+key2=hello world
+key3=<empty string>
+```
+____
+
+## Failure Modes
+
+There can be multipe failure modes:
+* The key/interpolated value do not meet the [AWS Tag Requirements](https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html)
+* The key is not allowed (such as keys used internally by the CSI driver e.g., 'CSIVolumeName').
+In these cases, the CSI driver will not provision a volume, but instead return an error.
```

Comment:
Is this currently the case?

Reply:
I'm not sure. I think we currently append the tag to the list of tags even if the tag key already exists. In that case, I'd imagine our API returns an error, but I haven't tested it.

Reply:
Confirmed via testing that this properly throws an error.